internal/wire/wire.go

```diff
@@ -315,11 +364,84 @@ func viewIDs(v *view.View, path *path.Path, key string, graphDB *cayley.Handle)
 		ids = append(ids, key)
 	}
 
-	return ids, nil
+	return ids, embeds, nil
+}
+
+// extractEmbeddedRel extracts a relationship that needs to be embedded on a view item.
+func extractEmbeddedRels(v *view.View, taggedIDs map[string]relList, tagOrder map[string]string, key string) (embeddedRels, error) {
+
+	// embeds will contain the indication of the embedded relationships.
+	var embeds embeddedRels
+
+	// Loop over the taggedIDs determining the embedding based on the ordering
+	// of the view path.
+	for tag, rels := range taggedIDs {
+
+		// Get the order of the tagged ID.
+		a := tagOrder[tag]
+		aliasOrder := strings.Split(a, "_")
+		order, err := strconv.Atoi(aliasOrder[len(aliasOrder)-1])
+		if err != nil {
+			return nil, err
+		}
+
+		// If the order is greater than 1, we should embed at the next tagged
+		// level up from the item.
+		var embedTag string
+		var ok bool
+		if order > 1 {
+
+			// Get the tag of the item in which the embed should be placed.
+			for order > 1 {
+				alias := aliasOrder[0] + "_" + strconv.Itoa(order-1)
```

Comment:
Should still check that aliasOrder has a length greater than 0, `strings.Split` can still return an empty slice.